File: Demo4Angular/src/app/authority/models/api-resource.ts
export class Property {
    id: number;
    key: string;
    value: string;
}

export class Secret {
    id: number;
    description: string;
    value: string;
    expiration?: Date;
    type: string;
    created: Date;
    constructor() {
        this.description = '';
        this.value = '';
        this.type = 'cc';
        this.created = new Date();
    }
}

export class UserClaim {
    id: number;
    type: string;
}

export class ApiSecret extends Secret {
    apiResourceId: number;
    apiResource: ApiResource;
    constructor(parent: ApiResource) {
        super();
        this.id = 1;
        this.apiResourceId = parent.id;
        this.apiResource = parent;
        const sorted = parent.secrets.sort((a, b) => b.id - a.id);
        if (sorted && sorted.length > 0) {
            this.id = sorted[0].id + 1;
        }
    }
}

export class ApiScope {
    id: number;
    name: string;
    displayName: string;
    description: string;
    required: boolean;
    emphasize: boolean;
    showInDiscoveryDocument: boolean;
    userClaims: Array<ApiScopeClaim>;
    apiResourceId: number;
    apiResource: ApiResource;
}

export class ApiScopeClaim extends UserClaim {
    apiScopeId: number;
    apiScope: ApiScope;
}

export class ApiResourceClaim extends UserClaim {
    apiResourceId: number;
    apiResource: ApiResource;
}

export class ApiResourceProperty extends Property {
    apiResourceId: number;
    apiResource: ApiResource;
}

export class ApiResource {
    id: number;
    enabled: boolean;
    name: string;
    displayName: string;
    description: string;
    secrets: Array<ApiSecret>;
    scopes: Array<ApiScope>;
    userClaims: Array<ApiResourceClaim>;
    properties: Array<ApiResourceProperty>;
    created: Date;
    updated: Date;
    lastAccessed: Date;
    nonEditable: boolean;

    constructor() {
        this.id = 0;
        this.enabled = true;
        this.secrets = [];
        this.scopes = [];
        this.userClaims = [];
        this.properties = [];
    }
}

File: Demo4DotNetCore/Demo4DotNetCore.ResourceServer/Areas/Identity/Model/ApiResource.cs
using Demo4DotNetCore.ResourceServer.Model;
using System;
using System.Collections.Generic;

namespace Demo4DotNetCore.ResourceServer.Identity.Model
{
    public class ApiResource : BaseModel<int>
    {
        public bool Enabled { get; set; }
        public string Name { get; set; }
        public string DisplayName { get; set; }
        public string Description { get; set; }
        public List<ApiSecret> Secrets { get; set; }
        public List<ApiScope> Scopes { get; set; }
        public List<ApiResourceClaim> UserClaims { get; set; }
        public List<ApiResourceProperty> Properties { get; set; }
        public DateTime Created { get; set; }
        public DateTime? Updated { get; set; }
        public DateTime? LastAccessed { get; set; }
        public bool NonEditable { get; set; }
    }

    public class Secret : BaseModel<int>
    {
        public string Description { get; set; }
        public string Value { get; set; }
        public DateTime? Expiration { get; set; }
        public string Type { get; set; }
        public DateTime Created { get; set; }
    }

    public class ApiSecret : Secret
    {
        public int ApiResourceId { get; set; }
        public ApiResource ApiResource { get; set; }
    }

    public class UserClaim : BaseModel<int>
    {
        public string Type { get; set; }
    }

    public class ApiScopeClaim : UserClaim
    {
        public int ApiScopeId { get; set; }
        public ApiScope ApiScope { get; set; }
    }

    public class ApiScope : BaseModel<int>
    {
        public string Name { get; set; }
        public string DisplayName { get; set; }
        public string Description { get; set; }
        public bool Required { get; set; }
        public bool Emphasize { get; set; }
        public bool ShowInDiscoveryDocument { get; set; }
        public List<ApiScopeClaim> UserClaims { get; set; }
        public int ApiResourceId { get; set; }
        public ApiResource ApiResource { get; set; }
    }

    public class ApiResourceClaim : UserClaim
    {
        public int ApiResourceId { get; set; }
        public ApiResource ApiResource { get; set; }
    }

    public class Property : BaseModel<int>
    {
        public string Key { get; set; }
        public string Value { get; set; }
    }

    public class ApiResourceProperty : Property
    {
        public int ApiResourceId { get; set; }
        public ApiResource ApiResource { get; set; }
    }
}
